### PR TITLE
Update to signing key with 2023 expiry

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review

## Test plan

General verification:
- [ ] Confirm that the key in this PR corresponds to the updated SecureDrop release key with 2023 expiry. https://securedrop.org/securedrop-release-key.asc

On Qubes 4.0:
- If you have an existing install:
  - Run the command `gpg /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation`.
  -  [ ]  Confirm that you're either seeing the test key, or a key with the 2022 expiry.
  - For bonus points, double-check that the key isn't hiding in RPM's keyring with the command `rpm -q gpg-pubkey --qf '%{NAME}-%{VERSION}-%{RELEASE}\t%{SUMMARY}\n'`.
- `make clone` from this branch to build an RPM
- Install this RPM manually using `sudo dnf install` (it will be generated in `rpm-build/RPMS/noarch` after `make clone`).
- Configure your environment in `/usr/share/securedrop-workstation-dom0-config/`. Ensure the environment `config.json` is set to `prod`
- Run `sdw-admin --apply`.
- Repeat the `gpg` verification from earlier.
- [ ] Confirm that you're now seeing the updated key.
- [ ] Confirm that `make test` still runs successfully or that no new tests are failing.

On Qubes 4.1:
- This won't be testable until #784 lands. After that, the test procedure should be identical.

### CODE REVIEW

**Comment by eloquence** [comment]
So, a note on testing:
- For Qubes 4.0, I would recommend testing using the backport PR, https://github.com/freedomofpress/securedrop-workstation/pull/788. This ensures we're only testing the signing key change relative to the last prod RPM release.
- For Qubes 4.1 testing, this is blocked on merging #784, as `sdw-admin --apply` will not successfully run on 4.1 until then.

**Review by creviera — APPROVED** [review]
I tested this for Qubes 4.0 (using https://github.com/freedomofpress/securedrop-workstation/pull/788) and will run through the test plan again for Qubes 4.1 after merging this PR into the `600-bullseye-template-and-repo-magic` branch (will update that PR with the test plan accordingly).

## Test Plan

On Qubes 4.0:
- If you have an existing install:
  - Run the command `gpg /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation`.
  - [x]  Confirm that you're either seeing the test key, or a key with the 2022 expiry.
        > It was the test key
  - [x] For bonus points, double-check that the key isn't hiding in RPM's keyring with the command `rpm -q gpg-pubkey --qf '%{NAME}-%{VERSION}-%{RELEASE}\t%{SUMMARY}\n'`.
        > I saw the test signing key, release signing key that expired 2021, and release signing key that expires in 2022 - did not see the new key that expires 2023
- `make clone` from this branch to build an RPM
- Install this RPM manually using `sudo dnf install` (it will be generated in `rpm-build/RPMS/noarch` after `make clone`).
- Configure your environment in `/usr/share/securedrop-workstation-dom0-config/`. Ensure the environment `config.json` is set to `prod`
- [x] Run `sdw-admin --apply`.
- Repeat the `gpg` verification from earlier.
- [x] Confirm that you're now seeing the updated key.
- [x] Confirm that `make test` still runs successfully or that no new tests are failing.

**Pull request merged**
